src/commands/createNewProject/javaSteps/JavaVersions.ts

```diff
@@ -0,0 +1,69 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as fse from 'fs-extra';
+import { localize } from '../../../localize';
+import { cpUtils } from "../../../utils/cpUtils";
+
+import path = require("path");
+
+export async function getJavaVersion(): Promise<number> {
+    const javaHome: string | undefined = process.env['JAVA_HOME'];
+    let javaVersion = javaHome ? await checkVersionInReleaseFile(javaHome) : undefined;
+    if (!javaVersion) {
+        javaVersion = await checkVersionByCLI(javaHome ? path.join(javaHome, 'bin', 'java') : 'java');
+    }
+    if (!javaVersion) {
+        const message: string = localize('javaNotFound', 'Failed to get java version, please ensure that java is installed and JAVA_HOME is set correctly.');
+        throw Error(message);
+    }
+    return javaVersion;
+}
+
+async function checkVersionInReleaseFile(javaHome: string): Promise<number | undefined> {
+    if (!javaHome) {
+        return undefined;
+    }
+    const releaseFile = path.join(javaHome, "release");
+    if (!fse.existsSync(releaseFile)) {
```

Comment:
Please use `AzExtFsUtils` for any fs calls. It uses `vscode.workspace.fs` so that fs calls are compatible for Codespaces and Virtual Workspaces.